src/docfx/build/page/Markup.cs

```diff
@@ -99,7 +92,7 @@ public static (MarkdownDocument ast, MarkupResult result) Parse(string content)
                     t_dependencyMap = dependencyMap;
                     t_bookmarkValidator = bookmarkValidator;
                     t_buildChild = buildChild;
-                    var html = Markdown.ToHtml(markdown, s_pipelineMapping[pipelineType]);
+                    var html = Markdown.ToHtml(markdown, SetupPipeline(pipelineType, xrefMap));
```

Comment:
don't create a new Pipeline each time (too slow). Use a thread static like `t_dependencyMap`.

Reply:
+1

---
In reply to: [218283381](https://github.com/dotnet/docfx/pull/3428#discussion_r218283381) [](ancestors = 218283381)